apps/emqx_resource/test/emqx_resource_SUITE.erl

```diff
@@ -3017,3 +3025,19 @@ trace_between_span(Trace0, Marker) ->
     {Trace1, [_ | _]} = ?split_trace_at(#{?snk_kind := Marker, ?snk_span := {complete, _}}, Trace0),
     {[_ | _], [_ | Trace2]} = ?split_trace_at(#{?snk_kind := Marker, ?snk_span := start}, Trace1),
     Trace2.
+
+wait_until_all_marked_as_retriable(NumExpected) when NumExpected =< 0 ->
+    ok;
+wait_until_all_marked_as_retriable(NumExpected) ->
+    Res = ?block_until(#{?snk_kind := buffer_worker_async_agent_down}, 10_000),
```

Comment:
Oops, noticed that this is not quite correct, as I might be double counting the same event...